File: 347. Top K Frequent Elements/main.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        return (i for i, _ in collections.Counter(nums).most_common(k))
        
#  Runtime: 185 ms
#  Memory: 18.6 MB
# O(n)

File: 347. Top K Frequent Elements/main.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
    unordered_map<int, int> hashmap;
    vector<int> ret_vec;
    priority_queue<pair<int, int>> pq;

    for(int num : nums){
        hashmap[num]++;
    }
    for (auto& kv: hashmap){
        pq.push(make_pair(kv.second, kv.first));
    }

    for(int i = 0; i< k; i++){
        ret_vec.push_back(pq.top().second);
        pq.pop();
    }

    return ret_vec;

    }
};

// O(n)
// Runtime: 11 ms
// Memory :13.8 MB
